src/Main.elm

```diff
@@ -16,25 +17,106 @@ main =
 
 
 type alias Model =
-    {}
+    { display : String
+    , displayStatus : DisplayStatus
+    , decimalStatus : DecimalStatus
+    }
 
 
 init : Model
 init =
-    {}
+    { display = "0"
+    , displayStatus = Locked
+    , decimalStatus = NoDecimal
+    }
 
 
 
 -- Update
 
 
 type Msg
-    = Message
+    = UpdateDisplay String
+
+
+type InputType
+    = Num
+    | Decimal
+
+
+type DisplayStatus
+    = Locked
+    | Unlocked
```

Comment:
What does it mean to have a `DisplayStatus` of `Locked`?

Reply:
Changed the names to better show if the display is showing the results of something.